src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs

```diff
@@ -65,7 +66,9 @@ public override void VisitGenericName(GenericNameSyntax node)
 
         public override void VisitIdentifierName(IdentifierNameSyntax node)
         {
-            if (node.IsKind(SyntaxKind.IdentifierName) && TrySimplify(node))
+            // Don't bother looking at the right side of A.B or A::B.  We will process those in
+            // one of our other top level Visit methods (like VisitQualifiedName).
+            if (node.IsKind(SyntaxKind.IdentifierName) && !node.IsRightSideOfDotOrArrowOrColonColon() && TrySimplify(node))
```

Comment:
```suggestion
            if (!node.IsRightSideOfDotOrArrowOrColonColon() && TrySimplify(node))
```

Reply:
ah, this still breaks something.  instead of IsRightSideOfDotOrArrowOrColonColon, looks like we need to start with basically asking if this is the name of a MemberAccessExpression.  

Note: i really despise some of these extensions.  `IsRightSideOfDot` is usually not something that should be asked.  versus "is name of member access" or "is name of qualified name".  etc. etc.

Reply:
IsRightSideOfDotOrArrowOrColonColon was fine, but I needed to update the code to handle `Attribute` suffixes separately. This is now fixed.